modules/replicator/src/main/java/org/apache/ignite/internal/replicator/ReplicaManager.java

```diff
@@ -235,7 +235,9 @@ private void onReplicaMessageReceived(NetworkMessage message, String senderConsi
             // replicaFut is always completed here.
             Replica replica = replicaFut.join();
 
-            CompletableFuture<?> result = replica.processRequest(request);
+            String senderId = clusterNetSvc.topologyService().getByConsistentId(senderConsistentId).id();
+
+            CompletableFuture<?> result = replica.processRequest(request, senderId);
```

Comment:
I don't like this way because we have no guarantee that the consistent id for the sender of the request matches the node id.

Reply:
added todo: IGNITE-20296